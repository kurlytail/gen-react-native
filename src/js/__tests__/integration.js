import { execSync } from 'child_process';

describe('# sgen-react-native integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        const output = execSync('./scripts/sgen-react-native.sh -h').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design', () => {
        const output = execSync(
            './scripts/sgen-react-native.sh -d src/test/fixture/design.js -o testoutput'
        ).toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design with merge', () => {
        let output = execSync(
            './scripts/sgen-react-native.sh -d src/test/fixture/design.js -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync(
            './scripts/sgen-react-native.sh -d src/test/fixture/design.js -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design and run react-native commands', () => {
        let output = execSync(
            './scripts/sgen-react-native.sh -d src/test/fixture/design.js -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync('npm install', { cwd: 'testoutput' }).toString();
        output = execSync('npm run lint', { cwd: 'testoutput' }).toString();
    });
});

describe('# sgen-react-native-nav integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        const output = execSync('./scripts/sgen-react-native-nav.sh -h').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design', () => {
        const output = execSync(
            './scripts/sgen-react-native-nav.sh -d src/test/fixture/nav.json -o testoutput'
        ).toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design with merge', () => {
        let output = execSync(
            './scripts/sgen-react-native-nav.sh -d src/test/fixture/nav.json -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync(
            './scripts/sgen-react-native-nav.sh -d src/test/fixture/nav.json -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design and run react-native commands', () => {
        let output = execSync(
            './scripts/sgen-react-native-nav.sh -d src/test/fixture/nav.json -o testoutput --overwrite=merge'
        ).toString();
        expect(output).toMatchSnapshot();
        output = execSync('npm install', { cwd: 'testoutput' }).toString();
        output = execSync('npm run lint', { cwd: 'testoutput' }).toString();
    });
});