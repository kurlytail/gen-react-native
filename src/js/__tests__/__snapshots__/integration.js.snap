// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design 1`] = `
"info: Generating testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generating testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: Generating testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generating testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generating testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: Generating testoutput/package.json from template templates/package.json._
info: Generating testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: Generating testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generating testoutput/src/js/index.js from template node_modules/@kurlytail/gen-npm/templates/index.js._
info: Generating testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generating testoutput/.babelrc from template templates/.babelrc._
info: Generating testoutput/app.json from template templates/app.json._
info: Generating testoutput/App.js from template templates/App.js._
info: Generating testoutput/index.js from template templates/index.js._
info: Generating testoutput/android/app/proguard-rules.pro from template templates/android/app/proguard-rules.pro._
info: Generating testoutput/android/app/BUCK from template templates/android/app/BUCK._
info: Generating testoutput/android/app/build.gradle from template templates/android/app/build.gradle._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/values/styles.xml from template templates/android/app/src/main/res/values/styles.xml._
info: Generating testoutput/android/app/src/main/res/values/strings.xml from template templates/android/app/src/main/res/values/strings.xml._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/AndroidManifest.xml from template templates/android/app/src/main/AndroidManifest.xml._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainApplication.java from template templates/android/app/src/main/java/com/projectname/MainApplication.java._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainActivity.java from template templates/android/app/src/main/java/com/projectname/MainActivity.java._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.jar from template templates/android/gradle/wrapper/gradle-wrapper.jar._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.properties from template templates/android/gradle/wrapper/gradle-wrapper.properties._
info: Generating testoutput/android/gradlew from template templates/android/gradlew._
info: Generating testoutput/android/keystores/debug.keystore.properties from template templates/android/keystores/debug.keystore.properties._
info: Generating testoutput/android/keystores/BUCK from template templates/android/keystores/BUCK._
info: Generating testoutput/android/build.gradle from template templates/android/build.gradle._
info: Generating testoutput/android/gradle.properties from template templates/android/gradle.properties._
info: Generating testoutput/android/gradlew.bat from template templates/android/gradlew.bat._
info: Generating testoutput/android/settings.gradle from template templates/android/settings.gradle._
info: Generating testoutput/ios/testoutput/AppDelegate.h from template templates/ios/PROJECTNAME/AppDelegate.h._
info: Generating testoutput/ios/testoutput/Base.lproj/LaunchScreen.xib from template templates/ios/PROJECTNAME/Base.lproj/LaunchScreen.xib._
info: Generating testoutput/ios/testoutput/main.m from template templates/ios/PROJECTNAME/main.m._
info: Generating testoutput/ios/testoutput/Images.xcassets/AppIcon.appiconset/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/AppIcon.appiconset/Contents.json._
info: Generating testoutput/ios/testoutput/Images.xcassets/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/Contents.json._
info: Generating testoutput/ios/testoutput/AppDelegate.m from template templates/ios/PROJECTNAME/AppDelegate.m._
info: Generating testoutput/ios/testoutput/Info.plist from template templates/ios/PROJECTNAME/Info.plist._
info: Generating testoutput/ios/testoutputTests/TESTOUTPUTTests.m from template templates/ios/PROJECTNAMETests/TESTOUTPUTTests.m._
info: Generating testoutput/ios/testoutputTests/Info.plist from template templates/ios/PROJECTNAMETests/Info.plist._
info: Generating testoutput/ios/testoutput-tvOSTests/Info.plist from template templates/ios/PROJECTNAME-tvOSTests/Info.plist._
info: Generating testoutput/ios/testoutput.xcodeproj/project.pbxproj from template templates/ios/PROJECTNAME.xcodeproj/project.pbxproj._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput-tvOS.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT-tvOS.xcscheme._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT.xcscheme._
info: Generating testoutput/ios/testoutput-tvOS/Info.plist from template templates/ios/PROJECTNAME-tvOS/Info.plist._
"
`;

exports[`# integration test ## should generate design and run react-native commands 1`] = `
"info: Generating testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generating testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: Generating testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generating testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generating testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: Generating testoutput/package.json from template templates/package.json._
info: Generating testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: Generating testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generating testoutput/src/js/index.js from template node_modules/@kurlytail/gen-npm/templates/index.js._
info: Generating testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generating testoutput/.babelrc from template templates/.babelrc._
info: Generating testoutput/app.json from template templates/app.json._
info: Generating testoutput/App.js from template templates/App.js._
info: Generating testoutput/index.js from template templates/index.js._
info: Generating testoutput/android/app/proguard-rules.pro from template templates/android/app/proguard-rules.pro._
info: Generating testoutput/android/app/BUCK from template templates/android/app/BUCK._
info: Generating testoutput/android/app/build.gradle from template templates/android/app/build.gradle._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/values/styles.xml from template templates/android/app/src/main/res/values/styles.xml._
info: Generating testoutput/android/app/src/main/res/values/strings.xml from template templates/android/app/src/main/res/values/strings.xml._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/AndroidManifest.xml from template templates/android/app/src/main/AndroidManifest.xml._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainApplication.java from template templates/android/app/src/main/java/com/projectname/MainApplication.java._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainActivity.java from template templates/android/app/src/main/java/com/projectname/MainActivity.java._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.jar from template templates/android/gradle/wrapper/gradle-wrapper.jar._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.properties from template templates/android/gradle/wrapper/gradle-wrapper.properties._
info: Generating testoutput/android/gradlew from template templates/android/gradlew._
info: Generating testoutput/android/keystores/debug.keystore.properties from template templates/android/keystores/debug.keystore.properties._
info: Generating testoutput/android/keystores/BUCK from template templates/android/keystores/BUCK._
info: Generating testoutput/android/build.gradle from template templates/android/build.gradle._
info: Generating testoutput/android/gradle.properties from template templates/android/gradle.properties._
info: Generating testoutput/android/gradlew.bat from template templates/android/gradlew.bat._
info: Generating testoutput/android/settings.gradle from template templates/android/settings.gradle._
info: Generating testoutput/ios/testoutput/AppDelegate.h from template templates/ios/PROJECTNAME/AppDelegate.h._
info: Generating testoutput/ios/testoutput/Base.lproj/LaunchScreen.xib from template templates/ios/PROJECTNAME/Base.lproj/LaunchScreen.xib._
info: Generating testoutput/ios/testoutput/main.m from template templates/ios/PROJECTNAME/main.m._
info: Generating testoutput/ios/testoutput/Images.xcassets/AppIcon.appiconset/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/AppIcon.appiconset/Contents.json._
info: Generating testoutput/ios/testoutput/Images.xcassets/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/Contents.json._
info: Generating testoutput/ios/testoutput/AppDelegate.m from template templates/ios/PROJECTNAME/AppDelegate.m._
info: Generating testoutput/ios/testoutput/Info.plist from template templates/ios/PROJECTNAME/Info.plist._
info: Generating testoutput/ios/testoutputTests/TESTOUTPUTTests.m from template templates/ios/PROJECTNAMETests/TESTOUTPUTTests.m._
info: Generating testoutput/ios/testoutputTests/Info.plist from template templates/ios/PROJECTNAMETests/Info.plist._
info: Generating testoutput/ios/testoutput-tvOSTests/Info.plist from template templates/ios/PROJECTNAME-tvOSTests/Info.plist._
info: Generating testoutput/ios/testoutput.xcodeproj/project.pbxproj from template templates/ios/PROJECTNAME.xcodeproj/project.pbxproj._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput-tvOS.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT-tvOS.xcscheme._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT.xcscheme._
info: Generating testoutput/ios/testoutput-tvOS/Info.plist from template templates/ios/PROJECTNAME-tvOS/Info.plist._
"
`;

exports[`# integration test ## should generate design with merge 1`] = `
"info: Generating testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generating testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: Generating testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generating testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generating testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: Generating testoutput/package.json from template templates/package.json._
info: Generating testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: Generating testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generating testoutput/src/js/index.js from template node_modules/@kurlytail/gen-npm/templates/index.js._
info: Generating testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generating testoutput/.babelrc from template templates/.babelrc._
info: Generating testoutput/app.json from template templates/app.json._
info: Generating testoutput/App.js from template templates/App.js._
info: Generating testoutput/index.js from template templates/index.js._
info: Generating testoutput/android/app/proguard-rules.pro from template templates/android/app/proguard-rules.pro._
info: Generating testoutput/android/app/BUCK from template templates/android/app/BUCK._
info: Generating testoutput/android/app/build.gradle from template templates/android/app/build.gradle._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/values/styles.xml from template templates/android/app/src/main/res/values/styles.xml._
info: Generating testoutput/android/app/src/main/res/values/strings.xml from template templates/android/app/src/main/res/values/strings.xml._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/AndroidManifest.xml from template templates/android/app/src/main/AndroidManifest.xml._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainApplication.java from template templates/android/app/src/main/java/com/projectname/MainApplication.java._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainActivity.java from template templates/android/app/src/main/java/com/projectname/MainActivity.java._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.jar from template templates/android/gradle/wrapper/gradle-wrapper.jar._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.properties from template templates/android/gradle/wrapper/gradle-wrapper.properties._
info: Generating testoutput/android/gradlew from template templates/android/gradlew._
info: Generating testoutput/android/keystores/debug.keystore.properties from template templates/android/keystores/debug.keystore.properties._
info: Generating testoutput/android/keystores/BUCK from template templates/android/keystores/BUCK._
info: Generating testoutput/android/build.gradle from template templates/android/build.gradle._
info: Generating testoutput/android/gradle.properties from template templates/android/gradle.properties._
info: Generating testoutput/android/gradlew.bat from template templates/android/gradlew.bat._
info: Generating testoutput/android/settings.gradle from template templates/android/settings.gradle._
info: Generating testoutput/ios/testoutput/AppDelegate.h from template templates/ios/PROJECTNAME/AppDelegate.h._
info: Generating testoutput/ios/testoutput/Base.lproj/LaunchScreen.xib from template templates/ios/PROJECTNAME/Base.lproj/LaunchScreen.xib._
info: Generating testoutput/ios/testoutput/main.m from template templates/ios/PROJECTNAME/main.m._
info: Generating testoutput/ios/testoutput/Images.xcassets/AppIcon.appiconset/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/AppIcon.appiconset/Contents.json._
info: Generating testoutput/ios/testoutput/Images.xcassets/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/Contents.json._
info: Generating testoutput/ios/testoutput/AppDelegate.m from template templates/ios/PROJECTNAME/AppDelegate.m._
info: Generating testoutput/ios/testoutput/Info.plist from template templates/ios/PROJECTNAME/Info.plist._
info: Generating testoutput/ios/testoutputTests/TESTOUTPUTTests.m from template templates/ios/PROJECTNAMETests/TESTOUTPUTTests.m._
info: Generating testoutput/ios/testoutputTests/Info.plist from template templates/ios/PROJECTNAMETests/Info.plist._
info: Generating testoutput/ios/testoutput-tvOSTests/Info.plist from template templates/ios/PROJECTNAME-tvOSTests/Info.plist._
info: Generating testoutput/ios/testoutput.xcodeproj/project.pbxproj from template templates/ios/PROJECTNAME.xcodeproj/project.pbxproj._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput-tvOS.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT-tvOS.xcscheme._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT.xcscheme._
info: Generating testoutput/ios/testoutput-tvOS/Info.plist from template templates/ios/PROJECTNAME-tvOS/Info.plist._
"
`;

exports[`# integration test ## should generate design with merge 2`] = `
"info: Generating testoutput/.editorconfig from template node_modules/@kurlytail/gen-npm/templates/.editorconfig._
info: Generating testoutput/.eslintignore from template node_modules/@kurlytail/gen-npm/templates/.eslintignore._
info: Generating testoutput/.eslintrc.js from template node_modules/@kurlytail/gen-npm/templates/.eslintrc.js._
info: Generating testoutput/.gitignore from template templates/.gitignore._
info: Generating testoutput/.npmignore from template node_modules/@kurlytail/gen-npm/templates/.npmignore._
info: Generating testoutput/.prettierrc from template node_modules/@kurlytail/gen-npm/templates/.prettierrc._
info: Generating testoutput/Jenkinsfile from template node_modules/@kurlytail/gen-npm/templates/Jenkinsfile._
info: Generating testoutput/package.json from template templates/package.json._
info: Generating testoutput/README.md from template node_modules/@kurlytail/gen-npm/templates/README.md._
info: Generating testoutput/webpack.config.js from template node_modules/@kurlytail/gen-npm/templates/webpack.config.js._
info: Generating testoutput/src/js/index.js from template node_modules/@kurlytail/gen-npm/templates/index.js._
info: Generating testoutput/src/js/__tests__/integration.js from template node_modules/@kurlytail/gen-npm/templates/integration.js._
info: Generating testoutput/.babelrc from template templates/.babelrc._
info: Generating testoutput/app.json from template templates/app.json._
info: Generating testoutput/App.js from template templates/App.js._
info: Generating testoutput/index.js from template templates/index.js._
info: Generating testoutput/android/app/proguard-rules.pro from template templates/android/app/proguard-rules.pro._
info: Generating testoutput/android/app/BUCK from template templates/android/app/BUCK._
info: Generating testoutput/android/app/build.gradle from template templates/android/app/build.gradle._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/res/values/styles.xml from template templates/android/app/src/main/res/values/styles.xml._
info: Generating testoutput/android/app/src/main/res/values/strings.xml from template templates/android/app/src/main/res/values/strings.xml._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher.png._
info: Generating testoutput/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png from template templates/android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png._
info: Generating testoutput/android/app/src/main/AndroidManifest.xml from template templates/android/app/src/main/AndroidManifest.xml._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainApplication.java from template templates/android/app/src/main/java/com/projectname/MainApplication.java._
info: Generating testoutput/android/app/src/main/java/com/projectname/MainActivity.java from template templates/android/app/src/main/java/com/projectname/MainActivity.java._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.jar from template templates/android/gradle/wrapper/gradle-wrapper.jar._
info: Generating testoutput/android/gradle/wrapper/gradle-wrapper.properties from template templates/android/gradle/wrapper/gradle-wrapper.properties._
info: Generating testoutput/android/gradlew from template templates/android/gradlew._
info: Generating testoutput/android/keystores/debug.keystore.properties from template templates/android/keystores/debug.keystore.properties._
info: Generating testoutput/android/keystores/BUCK from template templates/android/keystores/BUCK._
info: Generating testoutput/android/build.gradle from template templates/android/build.gradle._
info: Generating testoutput/android/gradle.properties from template templates/android/gradle.properties._
info: Generating testoutput/android/gradlew.bat from template templates/android/gradlew.bat._
info: Generating testoutput/android/settings.gradle from template templates/android/settings.gradle._
info: Generating testoutput/ios/testoutput/AppDelegate.h from template templates/ios/PROJECTNAME/AppDelegate.h._
info: Generating testoutput/ios/testoutput/Base.lproj/LaunchScreen.xib from template templates/ios/PROJECTNAME/Base.lproj/LaunchScreen.xib._
info: Generating testoutput/ios/testoutput/main.m from template templates/ios/PROJECTNAME/main.m._
info: Generating testoutput/ios/testoutput/Images.xcassets/AppIcon.appiconset/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/AppIcon.appiconset/Contents.json._
info: Generating testoutput/ios/testoutput/Images.xcassets/Contents.json from template templates/ios/PROJECTNAME/Images.xcassets/Contents.json._
info: Generating testoutput/ios/testoutput/AppDelegate.m from template templates/ios/PROJECTNAME/AppDelegate.m._
info: Generating testoutput/ios/testoutput/Info.plist from template templates/ios/PROJECTNAME/Info.plist._
info: Generating testoutput/ios/testoutputTests/TESTOUTPUTTests.m from template templates/ios/PROJECTNAMETests/TESTOUTPUTTests.m._
info: Generating testoutput/ios/testoutputTests/Info.plist from template templates/ios/PROJECTNAMETests/Info.plist._
info: Generating testoutput/ios/testoutput-tvOSTests/Info.plist from template templates/ios/PROJECTNAME-tvOSTests/Info.plist._
info: Generating testoutput/ios/testoutput.xcodeproj/project.pbxproj from template templates/ios/PROJECTNAME.xcodeproj/project.pbxproj._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput-tvOS.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT-tvOS.xcscheme._
info: Generating testoutput/ios/testoutput.xcodeproj/xcshareddata/xcschemes/testoutput.xcscheme from template templates/ios/PROJECTNAME.xcodeproj/xcshareddata/xcschemes/TESTOUTPUT.xcscheme._
info: Generating testoutput/ios/testoutput-tvOS/Info.plist from template templates/ios/PROJECTNAME-tvOS/Info.plist._
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
      --overwrite=ARG  Overwrite modes, can be one of skip, regen, error or merge                         (default: skip)
      --forceOverwrite Override overwrite flags in the generator map with the one provided on --overwrite
  -h, --help           display this help
"
`;
